cloudformation/lambda/AutoOffEC2.yaml

```diff
@@ -0,0 +1,104 @@
+                  logger.setLevel(logging.INFO)
+
+                  # define the connection
+                  ec2 = boto3.resource('ec2')
+
+
+                  def handler(event, context):
+                      # Use the filter() method of the instances collection to retrieve
+                      # all running EC2 instances.
+                      filters = [
+                          {
+                              'Name': 'instance-state-name',
+                              'Values': ['running']
+                          }
+                      ]
+
+                      # filter the instances
+                      instances = ec2.instances.filter(Filters=filters)
+                      stop_instances = set()
+                      prod_instances = {instance.id if prod_instance(instance) else stop_instances.add(instance.id) for instance in
+                                        instances}
+
+                      logger.info("All Instances")
+                      all_instances = prod_instances.union(stop_instances)
+                      logger.info(len(all_instances))
+                      logger.info(all_instances)
+
+                      logger.info("Production Instances")
+                      logger.info(len(prod_instances))
+                      logger.info(prod_instances)
+
+                      logger.info("Stopping Instances")
+                      logger.info(len(stop_instances))
+                      logger.info(stop_instances)
+
+                      if len(stop_instances) > 0:
+                          # perform the shutdown
+                          logger.info("Shutting down instances")
+                          logger.info(stop_instances)
+                          shutting_down = ec2.instances.filter(InstanceIds=stop_instances).stop()
+                          logger.info(shutting_down)
+
+
+                  def prod_instance(instance):
+                      if instance.tags is not None:
+                          for tag in instance.tags:
+                              logger.info(tag)
+                              if "Name" in tag['Key'] and "-aws-syd" in tag['Value']:
+                                  return True
+                              elif "elasticbeanstalk" in tag['Key']:
+                                  return True
+                              elif "AutoOff" in tag['Key'] and "False" in tag['Value']:
+                                  return True
+                          return False
+
+  LambdaExecutionRole:
+      Type: AWS::IAM::Role
+      Properties:
+          AssumeRolePolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+              - Effect: Allow
+                Principal: {Service: [lambda.amazonaws.com]}
+                Action: ['sts:AssumeRole']
+          Path: /
+          ManagedPolicyArns:
+          - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+          - arn:aws:iam::aws:policy/CloudWatchLogsFullAccess
```

Comment:
I'd get rid of this, until we know exactly what it needs, possibly nothing for logging